### PR TITLE
Better error msg if `len(predictions)` doesn't match `len(references)` in metrics

### PR DESCRIPTION
**Pull request by mariosasko**
Improve the error message in `Metric.add_batch` if `len(predictions)` doesn't match `len(references)`.

cc: @BramVanroy (feel free to test this code on your examples and review this PR)

### CODE REVIEW

File: src/datasets/metric.py [hunk 1]
```diff
@@ -435,13 +436,20 @@ def add_batch(self, *, predictions=None, references=None):
...
+            if match is not None:
+                error_msg = (
+                    f"Mismatch in the number of predictions ({match.group(1)}) and references ({match.group(2)})"
+                )
+            else:
+                error_msg = (
+                    f"Predictions and/or references don't match the expected format.\n"
+                    f"Expected format: {self.features},\n"
+                    f"Input predictions: {predictions},\n"
+                    f"Input references: {references}"
```
Comment (lhoestq) [inline src/datasets/metric.py:450]: Maybe let's not print the full list of predictions and references ? It could be very very long ^^'

**Review by lhoestq — APPROVED** [review]
Love it ! Thanks :)

Just one minor comment:

File: src/datasets/metric.py (hunk 1 above)
Comment (mariosasko) [reply src/datasets/metric.py:450]: Yes, I agree. We can implement a simple summarization similar to NumPy arrays. 

Currently, [NumPy](https://numpy.org/doc/stable/reference/generated/numpy.set_printoptions.html#numpy.set_printoptions) triggers summarization on arrays with more than 1000 elements. Do you have a threshold in mind for our use case?

Comment (lhoestq) [reply src/datasets/metric.py:450]: We can just crop at 5 elements IMO

Comment (mariosasko) [reply src/datasets/metric.py]: Let's make it 6 to have an equal number of elements on both sides of the summarized list if `len(batch) == 6`. 😃

**Comment by mariosasko** [comment]
I added a function, but it only works with the `list` type. For arrays/tensors, we delegate formatting to the frameworks.

**Review by lhoestq — APPROVED** [review]
Sounds good, thanks !

**Pull request merged**